charmbits/service/service_test.go

```diff
@@ -0,0 +1,129 @@
+package service_test
+
+import (
+	"time"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/gocharm/charmbits/service"
+	"github.com/juju/gocharm/hook"
+	"github.com/juju/gocharm/hook/hooktest"
+)
+
+type suite struct{}
+
+var _ = gc.Suite(&suite{})
+
+func (*suite) TestService(c *gc.C) {
+	var svc service.Service
+
+	startCount := 0
+	configChangedCount := 0
+	startService := func(ctxt *service.Context, args []string) (hook.Command, error) {
+		startCount++
+		c.Check(args, jc.DeepEquals, []string{"arg1", "arg2"})
+		return ctxt.ServeLocalRPC(TestRPCServer{})
+	}
+	r := &hooktest.Runner{
+		HookStateDir: c.MkDir(),
+		RegisterHooks: func(r *hook.Registry) {
+			svc.Register(r.Clone("svc"), "servicename", startService)
+			var ctxt *hook.Context
+			r.RegisterContext(func(hctxt *hook.Context) error {
+				ctxt = hctxt
+				return nil
+			}, nil)
+			r.RegisterHook("start", func() error {
+				return svc.Start("arg1", "arg2")
+			})
+			r.RegisterHook("stop", func() error {
+				return svc.Stop()
+			})
+			r.RegisterHook("config-changed", func() error {
+				var resp TestCallResponse
+				err := svc.Call("TestRPCServer.TestCall", &TestCallArg{"test"}, &resp)
+				c.Check(err, gc.IsNil)
+				c.Check(resp.Resp, gc.Equals, "test reply")
+				configChangedCount++
+				return nil
+			})
+		},
+		Logger: c,
+	}
+	notify := make(chan hooktest.ServiceEvent, 10)
+	service.NewService = hooktest.NewServiceFunc(r, notify)
+
+	err := r.RunHook("install", "", "")
+	c.Assert(err, gc.IsNil)
```

Comment:
You're already importing checkers. Is gc.IsNil rather than jc.ErrorIsNil a conscious decision you've made?